sisyphus/graph.py

```diff
@@ -112,6 +113,15 @@ def run_when_done(self, write_output=None):
                     # Set new link if needed
                     os.symlink(os.path.realpath(self._sis_path.get_path()), outfile_name)
                     logging.info("Finished output: %s" % outfile_name)
+
+                    if gs.FINISHED_LOG:
+                        if "/" in gs.FINISHED_LOG and not os.path.exists(os.path.dirname(gs.FINISHED_LOG)):
```

Comment:
You could also use:
```PY3
if os.path.dirname(gs.FINISHED_LOG):
  os.makedirs(os.path.dirname(gs.FINISHED_LOG)), exist_ok=True)
```

Reply:
Yes, I know, but my code should be more efficient:
* `dirname` will do some more stuff than just `"/" in gs.FINISHED_LOG`
* `makedirs` is optimized for the case that the dir does not exist yet. However, we will have the case that in almost all cases, the dir does already exist. In this case, `makedirs` has some overhead, and I wanted to avoid that.